test/collection.jl

```diff
@@ -0,0 +1,45 @@
+using CCDReduction: match_extension
+
+@testset "fitscollection" begin
+    # setting initial data
+    dir = joinpath(@__DIR__, "data")
+
+    # the two fits file used for testing have identical header keys
+    df = fitscollection(dir)
+    relevant_keys = filter(i -> i ∉ ("", "HISTORY"), keys(read_header(M6707HH[1])))
+    relevant_values_M6707HH = map(relevant_keys) do key
+        read_header(M6707HH[1])[key]
+    end
+    relevant_values_M35070V = map(relevant_keys) do key
+        read_header(M35070V[1])[key]
+    end
```

Comment:
I would be wary of calling `read_header` so many times- every single call is a read from disk, then getting one value from it. Compared to loading the headers once and iterating like so.